File: Light Code/VIQA/HtmlElements/SimpleElements/Link.cs
using OpenQA.Selenium;
using VIQA.HtmlElements.Interfaces;

namespace VIQA.HtmlElements
{
    public class Link : ClickableText, ILink
    {
        private const string LocatorTmpl = "a[{0}={1}]";
        public static string CommonLocatorById(string id) { return string.Format(LocatorTmpl, "id", id); }
        public static string CommonLocatorByNamed(string id) { return string.Format(LocatorTmpl, "name", id); }
        public static string CommonLocatorByClassName(string id) { return string.Format(LocatorTmpl, "class", id); }

        protected override string _typeName { get { return "Link"; } }

        public Link() { }
        public Link(string name) : base(name) { }
        public Link(string name, string cssSelector) : base(name, cssSelector) { }
        public Link(string name, By byLocator) : base(name, byLocator) { }
        public Link(string name, IWebElement webElement) : base(name, webElement) { }
        public Link(IWebElement webElement) : base(webElement) { }

        public string Reference
        {
            get { return DoVIAction("Get Reference", () => GetWebElement().GetAttribute("href"), href => "Get href of link '" + href + "'"); }
        }

    }
}

File: Java/src/main/java/ru/viqa/ui_testing/elements/simpleElements/Link.java
package ru.viqa.ui_testing.elements.simpleElements;

import ru.viqa.ui_testing.elements.interfaces.*;
import org.openqa.selenium.*;

import static java.lang.String.format;

/**
 * Created by 12345 on 02.10.2014.
 */
public class Link extends ClickableText implements ILink {
    private static final String LocatorTmpl = "a[%s=%s]";
    public static String commonLocatorById(String id) { return format(LocatorTmpl, "id", id); }
    public static String commonLocatorByNamed(String id) { return format(LocatorTmpl, "name", id); }
    public static String commonLocatorByClassName(String id) { return format(LocatorTmpl, "class", id); }

    public Link() throws Exception{ super(); TypeName = "Link";}
    public Link(String name) throws Exception { super(name); }
    public Link(String name, String cssSelector) throws Exception { super(name, cssSelector); }
    public Link(String name, By byLocator) throws Exception { super(name, byLocator); }
    public Link(By byLocator) throws Exception { super(byLocator); }
    public Link(By byClickLocator, By byTextLocator) throws Exception { super(byClickLocator, byTextLocator); }
    public Link(String name, WebElement webElement) throws Exception { super(name, webElement); }
    public Link(WebElement webElement) throws Exception { super(webElement); }

    protected String getReferenceAction() throws Exception { return getUniqueWebElement().getAttribute("href"); }

    public final String getReference() throws Exception {
        return doVIActionResult("Get Reference", this::getReferenceAction, href -> "Get href of link '" + href + "'");
    }
}
